doc/versioning_and_life_cycle.md

```diff
@@ -0,0 +1,72 @@
+    - It is generally recommended to put the preview features into CLI extension first and migrate it to official CLI after they are stable and GA. We can also consider moving some preview features from official CLI to CLI extensions if they are proper for moving.
+
+### States Transition
+
+#### Preview to GA
+Generally, if the service is stable and the design of the commands will not change, we are confident that these commands will not cause breaking changes in the future, then we can remove the `is_preview` flag and GA those commands.
+
+#### GA to Preview
+For the official CLI, it is generally not allowed to downgrade from GA to Preview, because the GA state means that the stability of commands has been promised to customers, and new instability cannot be introduced by downgrade to Preview. Moreover, since Azure CLI is an all-in-one package, users usually upgrade directly to the latest version of CLI. If users skip the intermediate version with those feature downgraded to Preview and directly upgrade to the latest version that has breaking changes, it will make a terrible experience for them.
+
+### Relationship with api-version
+
+At present, CLI sets the api-version according to the Resource Provider granularity, and there are corresponding configurations for different profiles. For most commands, we do not support customers to specify the requested api-version.
+
+- For the GA api-version, if the command itself determines the final design, then we usually make the implemented command also in the GA state.
+
+- For the public preview api-version, the situation is relatively complicated, and we usually divide it into the following cases:
+    - if those features are newly onboard RPs, command groups or extensions, we usually mark them as `is_preview`. 
+    - If there are some existing commands under these RPs, but only the new features need to be released through the preview api-version, and we need to add some new commands or parameters to support them, then we need to upragde to the preview api-version and only mark `is_preview` on those new commands and parameters, while the existing commands and parameters remain in the GA state. We still need to ensure that the existing commands and parameters are stable without breaking changes.
+    - Due to the higher flexibility of CLI extension, we plan to make the release status of extension consistent with the api-version status: if the api-version is preview, the whole extension should be in preview. And we need to add "preview" suffix in extension version number *(comply with [pep-0440](https://peps.python.org/pep-0440/))* to prevent customers from upgrading to preview version automatically. We will add support for `az extension add` and `az upgrade` to allow customers to install/upgrade to stable version as default behavior and to install/upgrade to preview version by specifying extra parameter in the future. 
```

Comment:
```suggestion
    - Due to the higher flexibility of CLI extension, we want customers to use the stable api-version for extension by default, so we plan to make the release status of extension consistent with the api-version status: if the api-version is preview, the whole extension should be in preview. And we need to add "preview" suffix in extension version number *(comply with [pep-0440](https://peps.python.org/pep-0440/))* to prevent customers from upgrading to preview version automatically. We will add support for `az extension add` and `az upgrade` to allow customers to install/upgrade to stable version as default behavior and to install/upgrade to preview version by specifying extra parameter in the future. 
```